solana-programs/anchor/audius-data/programs/audius-data/src/lib.rs

```diff
@@ -230,45 +224,27 @@ pub mod audius_data {
     /*
         Track related functions
     */
-    pub fn create_track(ctx: Context<CreateTrack>, metadata: String) -> Result<()> {
-        msg!("Audius::CreateTrack");
+    pub fn create_track(ctx: Context<CreateTrack>, _id: String, _metadata: String) -> Result<()> {
+        // Reject if update submitted with invalid user authority
         if ctx.accounts.authority.key() != ctx.accounts.user.authority {
             return Err(ErrorCode::Unauthorized.into());
         }
-        // Set owner to user storage account
-        ctx.accounts.track.owner = ctx.accounts.user.key();
-        ctx.accounts.track.track_id = ctx.accounts.audius_admin.track_id;
-        // Increment global track ID after assignment to this track in particular
-        // Ensures each track has a unique numeric ID
-        ctx.accounts.audius_admin.track_id = ctx.accounts.audius_admin.track_id + 1;
-        msg!("AudiusTrackMetadata = {:?}", metadata);
         Ok(())
     }
 
-    pub fn update_track(ctx: Context<UpdateTrack>, metadata: String) -> Result<()> {
-        msg!("Audius::UpdateTrack");
-        if ctx.accounts.user.key() != ctx.accounts.track.owner {
-            return Err(ErrorCode::Unauthorized.into());
-        }
+    pub fn update_track(ctx: Context<UpdateTrack>, _id: String, _metadata: String) -> Result<()> {
```

Comment:
now that all 3 of these are just pass through, I am going to consolidate them into a single function

Reply:
consolidated along w/EntityTypes - this will remove our need for a separate identical set of playlist functions